tools/src/test/java/org/apache/kafka/tools/other/ReplicationQuotasTestRig.java

```diff
@@ -0,0 +1,480 @@
+import scala.collection.Seq;
+
+import javax.imageio.ImageIO;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.TreeMap;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import java.util.function.IntSupplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static java.nio.file.StandardOpenOption.APPEND;
+import static java.nio.file.StandardOpenOption.CREATE;
+import static org.apache.kafka.test.TestUtils.DEFAULT_MAX_WAIT_MS;
+
+/**
+ * Test rig for measuring throttling performance. Configure the parameters for a set of experiments, then execute them
+ * and view the html output file, with charts, that are produced. You can also render the charts to the screen if
+ * you wish.
+ *
+ * Currently you'll need about 40GB of disk space to run these experiments (largest data written x2). Tune the msgSize
+ * & #partitions and throttle to adjust this.
+ */
+public class ReplicationQuotasTestRig {
+    public static final Logger LOGGER = LoggerFactory.getLogger(ReplicationQuotasTestRig.class);
+
+    public static final int K = 1000 * 1000;
+
+    private static final String DIR;
+
+    static {
+        PropertyConfigurator.configure("tools/src/test/resources/log4j-off.properties");
+
+        new File("Experiments").mkdir();
+        DIR = "Experiments/Run" + Long.valueOf(System.currentTimeMillis()).toString().substring(8);
+        new File(DIR).mkdir();
+    }
+
+    public static void main(String[] args) {
+        boolean displayChartsOnScreen = args.length > 0 && Objects.equals(args[0], "show-gui");
+        Journal journal = new Journal();
+
+        List<ExperimentDef> experiments = Arrays.asList(
+            //1GB total data written, will take 210s
+            new ExperimentDef("Experiment1", 5, 20, 1 * K, 500, 100 * 1000),
+            //5GB total data written, will take 110s
+            new ExperimentDef("Experiment2", 5, 50, 10 * K, 1000, 100 * 1000),
+            //5GB total data written, will take 110s
+            new ExperimentDef("Experiment3", 50, 50, 2 * K, 1000, 100 * 1000),
+            //10GB total data written, will take 110s
+            new ExperimentDef("Experiment4", 25, 100, 4 * K, 1000, 100 * 1000),
+            //10GB total data written, will take 80s
+            new ExperimentDef("Experiment5", 5, 50, 50 * K, 4000, 100 * 1000)
+        );
+        experiments.forEach(def -> run(def, journal, displayChartsOnScreen));
+
+        if (!displayChartsOnScreen)
+            Exit.exit(0, None$.empty());
```

Comment:
Why not `Option.empty()` here and at all places where empty Option is used?
`None$` looks ugly. Is it necessary?

Reply:
Great catch. Fixed.